core/dnsserver/config.go

```diff
@@ -53,6 +54,9 @@ type Config struct {
 	// TLSConfig when listening for encrypted connections (gRPC, DNS-over-TLS).
 	TLSConfig *tls.Config
 
+	// Timeouts for TCP, TLS and HTTPS servers.
+	Timeouts map[string]time.Duration
```

Comment:
```suggestion
	Timeouts map[TimeoutType]time.Duration
```

and create new enum type
```
type TimeoutType uint8

const (
	IdleTimeout TimeoutType = iota
	ReadTimeout
	WriteTimeout
)
```

Reply:
Could instead just be 3 separate properties in the struct. No need that I see for them to be in a map.